src/test/java/iudx/catalogue/server/mlayer/vocabulary/DataModelTest.java

```diff
@@ -0,0 +1,171 @@
+package iudx.catalogue.server.mlayer.vocabulary;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.*;
+
+import io.vertx.core.*;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.client.HttpRequest;
+import io.vertx.ext.web.client.HttpResponse;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import iudx.catalogue.server.database.ElasticClient;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.*;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+@ExtendWith(VertxExtension.class)
+public class DataModelTest {
+  private static final Logger LOGGER = LogManager.getLogger(DataModelTest.class);
+  MultiMap mockHeaders = mock(MultiMap.class);
+  @Mock private WebClient webClient;
+  @Mock private ElasticClient mockElasticClient;
+  @Mock private HttpResponse<Buffer> mockHttpResponse;
+  @Mock private HttpRequest<Buffer> mockHttpRequest;
+  @Mock private Buffer mockBuffer;
+  @InjectMocks private DataModel dataModel;
+
+  @BeforeEach
+  void setUp() {
+    MockitoAnnotations.openMocks(this);
+    dataModel = new DataModel(webClient, mockElasticClient, "test-index");
+  }
+
+  @Test
+  void testGetDataModelInfo_Success(VertxTestContext vertxTestContext) {
+    // Mocking Elasticsearch client's async search
+    when(mockElasticClient.searchAsync(anyString(), anyString(), any()))
+        .thenAnswer(
+            invocation -> {
+              ((Handler<AsyncResult<JsonObject>>) invocation.getArgument(2))
+                  .handle(
+                      Future.succeededFuture(
+                          new JsonObject()
```

Comment:
create JSON object first then pass it here

Reply:
Done